modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/OrganizationResourceImpl.java

```diff
@@ -219,22 +228,70 @@ public Organization postOrganization(Organization organization)
 	}
 
 	@Override
-	public void postUserAccountByEmailAddress(
+	public UserAccount postUserAccountByEmailAddress(
 			String organizationId, String emailAddress)
 		throws Exception {
 
-		_organizationService.addUserOrganizationByEmailAddress(
-			emailAddress, _getServiceBuilderOrganizationId(organizationId));
+		ServiceContext serviceContext = new ServiceContext();
+
+		serviceContext.setCompanyId(contextCompany.getCompanyId());
+		serviceContext.setLanguageId(
+			contextAcceptLanguage.getPreferredLanguageId());
+		serviceContext.setUserId(contextUser.getUserId());
+
+		User user = _organizationService.addOrganizationUserByEmailAddress(
+			emailAddress, _getServiceBuilderOrganizationId(organizationId),
+			serviceContext);
+
+		return _userResourceDTOConverter.toDTO(user);
 	}
 
 	@Override
-	public void postUserAccountsByEmailAddress(
-			String organizationId, String[] emailAddresses)
+	public Page<UserAccount> postUserAccountsByEmailAddress(
+			String organizationId, String organizationRoleIds,
+			String[] emailAddresses)
 		throws Exception {
 
+		List<UserAccount> userAccounts = new ArrayList<>();
+
 		for (String emailAddress : emailAddresses) {
-			postUserAccountByEmailAddress(organizationId, emailAddress);
+			userAccounts.add(
+				postUserAccountByEmailAddress(organizationId, emailAddress));
 		}
+
+		if (!organizationRoleIds.isEmpty()) {
+			String[] orgRoleIds = StringUtil.split(
```

Comment:
Rename to `organizationRoleIds`
```suggestion
			String[] organizationRoleIds = StringUtil.split(
```

Reply:
organizationRoleIds is already defined as the parameter of the method